src/player/player.c

```diff
@@ -488,7 +458,10 @@ void playerUpdateGrabbedObject(struct Player* player) {
         vector3Add(&player->lookTransform.position, &grabPoint, &grabPoint);
         grabPoint.y += grabY;
         
-        struct Quaternion grabRotation;
+        // remember delta between forwardRotation and lookTransform.rotation
+        struct Quaternion lookRotationDelta, forwardRotationInv;
+        quatConjugate(&forwardRotation, &forwardRotationInv);
+        quatMultiply(&forwardRotationInv, &player->lookTransform.rotation, &lookRotationDelta);
```

Comment:
> Can you do this calculation in `grab_rotation.c` since that's the only place it's needed? Just need to pass `player->lookTransform.rotation` into `grabRotationUpdate()` instead of the delta.

Yes, this is a bit suboptimal. Without the portal transform this would be trivial to just move to `grabRotationUpdate()`, but the delta needs to be calculated before the portal transform for it to be correct. Otherwise, I'd need to transform the `lookTransform.rotation` too, which is again not optimal to do in `grabRotationUpdate()` as it's a player function. I'll need to take a closer look at how to clean this up properly.

Reply:
Ah, good point. In that case I think keeping it here makes sense given the alternatives.

Reply:
Agreed